src/timesync/tcp_connect.c

```diff
@@ -0,0 +1,59 @@
+#include <stdlib.h>
+#include <sys/types.h>
+#include <sys/socket.h>
+#include <fcntl.h>
+#include <netdb.h>
+#include <string.h>
+#include <netinet/in.h>
+#include <netinet/tcp.h>
+#include <arpa/inet.h>
+#include <unistd.h>
+#include <endian.h>
+
+int NTS_attach_socket(const char *host, int port, int type);
+
+int NTS_attach_socket(const char *host, int port, int type) {
+        struct addrinfo hints;
+        memset(&hints, 0, sizeof(hints));
+        hints.ai_socktype = type;
+
+        struct addrinfo *info;
+        if (getaddrinfo(host, NULL, &hints, &info) != 0)
```

Comment:
please never use a blocking call like this in an event-loop based serice such as timesyncd. use sd_resolve_xyz